pkgs/desktops/kde-5/applications/akonadi.nix

```diff
@@ -0,0 +1,44 @@
+{ boost
+, ecm
+, kdeApp
+, kdesignerplugin
+, kitemmodels
+, lib
+, makeQtWrapper
+, mariadb
+, postgresql
+, shared_mime_info
+}:
+
+kdeApp {
+  name = "akonadi";
+  meta = {
+    description = "KDE PIM Storage Service";
+    license = with lib.licenses; [ gpl2 ];
+    homepage = http://pim.kde.org/akonadi;
+    maintainers = with lib.maintainers; [ vandenoever ];
+  };
+  nativeBuildInputs = [
+    ecm
+    makeQtWrapper
+  ];
+  buildInputs = [
+    boost
+    kdesignerplugin
+    kitemmodels
+  ];
+  propagatedBuildInputs = [
+    mariadb
+    postgresql
+  ];
+
+  postInstall = ''
+    wrapQtProgram "$out/bin/akonadi_agent_launcher"
```

Comment:
Use [`kdeWrapper`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/kde-5/applications/dolphin.nix#L27) instead of `wrapQtProgram`. The closure size will be smaller and all the necessary runtime dependencies will be found.